### ISSUE TITLE
Line chart series limit doesn't show correct result on chart

### ISSUE THREAD

**Issue by zuzana-vej** [apache/incubator-superset#11426]
The Series Limit dropdown in Superset Explore allows to limit the number of series when there are too many. However, it looks like the specified limit is not accurate. 
The query generated from the chart looks correct, and running the query in SQL Lab gives the expected result - if limit is 5 items, all 5 items are showing up there. So the problem seem to be with the chart.

### Expected results

Series Limit = 10 shows 10 results

### Actual results

Setting Series Limit = 10 gave me 9 results when 10 series actually existed. In another case series limit = 10 gave 14 results.  Note I believe when using with time shift, the time shifted lines might not be counting to the total limit (regardless of how this case is actually handled, current behavior has issue). 

#### Screenshots

Example 1: Returned 14 series, while limit is set to 10.
![Screen Shot 2020-10-26 at 8 00 04 AM](https://user-images.githubusercontent.com/61221714/97189554-f5eabb80-1761-11eb-92d3-e4ce306b6dcc.png)

Example 2: Returned 6 series while limit it set to 5.
![Screen Shot 2020-10-26 at 8 00 29 AM](https://user-images.githubusercontent.com/61221714/97189576-fc793300-1761-11eb-8322-76b6c1956d94.png)

Example 3: Even when removing time shift comparisons, the result is 2 series, while limit is set to 5. 
![Screen Shot 2020-10-26 at 8 20 44 AM](https://user-images.githubusercontent.com/61221714/97191652-53800780-1764-11eb-9585-8ce6c9834ecc.png)

#### How to reproduce the bug

1. Go to Chart Explore, Line chart
2. Click on Series and set the limit
3. See how many results you actually get

### Environment

(please complete the following information):

- superset version: `master`

### Checklist

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by junlincc** [comment]
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/67837651/97212719-06a92a80-177e-11eb-87a4-908a9ab98028.gif)

Hey Zuzana, I just tested on latest master, it looks alright to me. would you mind confirming?

**Comment by junlincc** [comment]
![3](https://user-images.githubusercontent.com/67837651/97214616-9354e800-1780-11eb-8fc8-51ab8f7e487b.gif)

Zuzana, I just tested on latest master, and it looks alright. can you double check? thanks

**Comment by zuzana-vej** [comment]
Okay.. I will check again. Thanks!

**Comment by villebro** [comment]
@zuzana-vej To help narrow this down, can you test this without time shift to make sure that's not causing trouble?

**Comment by zuzana-vej** [comment]
Yes I tested both with / without time series.And both have issue. Just tested on latest master now.

Here is an example of limit to 5 series:
<img width="809" alt="Screen Shot 2020-10-28 at 7 53 04 PM" src="https://user-images.githubusercontent.com/61221714/97519943-660e6280-1957-11eb-9405-f4a5b5ef3957.png">


Here is the same chart without the limit: 
<img width="809" alt="Screen Shot 2020-10-28 at 7 53 18 PM" src="https://user-images.githubusercontent.com/61221714/97519951-6870bc80-1957-11eb-8b8d-de375a167efa.png">

cc: @graceguo-supercat

**Comment by mistercrunch** [comment]
I'd be curious to see the generated SQL here

**Comment by zuzana-vej** [comment]
@mistercrunch SQL looks correct (limit is set up) and taking that SQL and running it in SQL Lab gives the intended result (e.g. there are the 5 unique items if limit is set to 5). 
We will look into it a bit more to troubleshoot.

**Comment by zuzana-vej** [comment]
looks like a sort issue rather than time series, closing this off.

**Comment by anilvpatel21** [comment]
@zuzana-vej I am also getting the same issue, did you find any solution to work around.

**Comment by zuzana-vej** [comment]
Yes, we determined it's not a bug. Here is when we found out of was happening and how to prevent it:
When there is no sort metric, so if you run query with LIMIT 10, it can limit kind of randomly. If you add "advanced analytics" with offset, it basically results in 2 queries and since there is no sort metric and they are limited to 10 (or whatever you use) the returned results could mismatch so you might see more than 10 values in the final chart. 

Easiest way to validate once you build your chart is to get the query and try to run it in SQL Lab. When we tried it without the sort metric, we could get different results different times the query is run. Adding the sort metric should fix the issue.
